lib/Mojolicious/Plugin/CGI.pm

```diff
@@ -278,15 +288,15 @@ sub _stdout_cb {
     $buf .= $chunk;
     $buf =~ s/^(.*?\x0a\x0d?\x0a\x0d?)//s or return;    # false until all headers has been read from the CGI script
     $headers = $1;
-
-    if ($headers =~ /^HTTP/) {
+    if ($headers =~ /^HTTP\/\d\.\d (\d\d\d)?/) {
+      $c->res->code($1) if $1;
```

Comment:
Could you do this instead:

    if ($headers =~ /^HTTP/) {
      $c->res->code($1) if $headers =~ m!^HTTP/\d\.\d (\d\d\d)!;

Not sure what will break if we make the $headers parsing more strict. Too bad there's no tests for this :disappointed: